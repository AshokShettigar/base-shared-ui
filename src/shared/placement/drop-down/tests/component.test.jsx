import { shallow } from 'enzyme';

import PlacementDropDown from '../component';

describe('PlacementDropDown component', () => {
    const requiredProps = {
        position: 'bottom-start',
        targetRef: document.createElement('div'),
        children: () => <div>Children</div>,
        onDismiss: jest.fn()
    };

    it('should render component with default props', () => {
        const wrapper = shallow(<PlacementDropDown {...requiredProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with class names', () => {
        const wrapper = shallow(
            <PlacementDropDown
                {...requiredProps}
                className="className"
                contentClassName="contentClassName"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('shouldn\'t render component', () => {
        const wrapper = shallow(<PlacementDropDown {...requiredProps} rendered={false} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render hidden component', () => {
        const wrapper = shallow(<PlacementDropDown {...requiredProps} hidden />);

        expect(wrapper).toMatchSnapshot();
    });
});
