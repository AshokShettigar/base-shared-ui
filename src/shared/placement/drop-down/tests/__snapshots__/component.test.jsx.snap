// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacementDropDown component should render component with class names 1`] = `
<PlacementDropDown
  className="className"
  contentClassName="contentClassName"
  context={null}
  hidden={false}
  onDismiss={[Function]}
  position="bottom-start"
  targetRef={<div />}
/>
`;

exports[`PlacementDropDown component should render component with default props 1`] = `
<PlacementDropDown
  context={null}
  hidden={false}
  onDismiss={[Function]}
  position="bottom-start"
  targetRef={<div />}
/>
`;

exports[`PlacementDropDown component should render hidden component 1`] = `
<div
  hidden={true}
>
  <PlacementDropDown
    context={null}
    hidden={true}
    onDismiss={[Function]}
    position="bottom-start"
    targetRef={<div />}
  />
</div>
`;

exports[`PlacementDropDown component shouldn't render component 1`] = `null`;
